distmap

```diff
@@ -235,31 +241,49 @@ GetOptions(
 	"disable-zipped-output" 		=>\$nozip,
 	"hadoop-home=s" 			=>\$hadoop_home,
 	"mappers-exe-archive=s" 		=>\$mappers_exe_archive,
-	
-	
-	"mapper=s" 				=> \@mapper,		# to check	
+
+
+	"mapper=s" 				=> \@mapper,		# to check
 	"mapper-path=s"				=> \@mapper_path,
 	"picard-jar=s"				=> \$picard_jar,
 	"picard-mergesamfiles-jar=s"		=> \$picard_mergesamfiles_jar,	# to ckeck
 	"picard-sortsam-jar=s"			=> \$picard_sortsam_jar,	# to check
+	"readtools=s"         => \$readtools,
 	"mapper-args=s"				=> \@mapper_args,
 	"bwa-sampe-args=s"			=> \$bwa_sampe_args,
-	"output-format=s"			=> \$output_format,					
+	"output-format=s"			=> \$output_format,
 	"job-desc=s"				=> \$job_desc,	# to check
 	"queue-name=s"				=> \$queue_name,	# to check
 	"gsnap-output-split"			=> \$gsnap_output_split,	# to check
 	"verbose"        		        => \$verbose,
 	"help"          		        =>\$help,
 	"h"          		        	=>\$help
-) or pod2usage(-msg=>"Wrong options",-verbose=>1); 
+) or pod2usage(-msg=>"Wrong options",-verbose=>1);
 
 pod2usage(-verbose=>2) if $help;
 
+if ( $hadoop_home ne "" ) {
+    # if $hadoop_home is set, make sure there is no trailing slash
+    $hadoop_home =~ s/\/$//;
+}
+else {
+    # try to find "hadoop" in path and prune "/bin/hadoop" from the full path
+    $hadoop_home = `which hadoop`;
+    $hadoop_home =~ s/\/bin\/hadoop$//;
+}
+
+if ( $readtools =~ /\.jar$/ ) {
+  $readtools = "java -Xmx8g -Dsnappy.disable=true -jar $readtools";
```

Comment:
I am not sure if setting the options here is useful. Different parts of DistMap might require different options, and this is a high constraint. Maybe a `ReadToolsUtils` package might be nice to handle all ReadTools related stuff:

* Version check: sub-rutine to check the version requirement for ReadTools, because it might change. That version can be a global variable and used everywhere before running the commands.
* ReadTools cmd constructor: sub-rutine to create a CMD for ReadTools. This will make more flexible the command line to tweak different parameters. The method will check if it is a jar file as here, and then provide the javaOpst in the correct way. Then, the rest of the CMD will be constructed. The parameters to that routine should be:
  - `$readtools`(as defined here)
  - __javaOpts__ for the -Xmx8g and -Dsnappy.disable=true
  - __toolName__ for `ReadsToDistmap` or `DownloadDistmapResult` (also `TrimReads` at some point). Like that, will be easier to realize if the program name is missing (as found before)
  - __arguments__ for the rest of the arguments

In the future, other sub-routines may be available when more functionality is added to ReadTools.

Reply:
The quicker solution which I now tried is to tweak the $readtools command for the jar constructed in distmap so that it also evaluates JAVA_OPTS.  When the tool is called, they can be set to whatever needed there.

Reply:
That's quick, but not that flexible and difficult. Anyway, I think that it should work for now and we can change it in further cleanup